scripts/upgrade

```diff
@@ -66,76 +83,57 @@ if [ -n "$is_public" ]; then
 	ynh_app_setting_delete --app=$app --key=is_public
 fi
 
-#=================================================
-# BACKUP BEFORE UPGRADE THEN ACTIVE TRAP
-#=================================================
-ynh_script_progression --message="Backing up the app before upgrading (may take a while)..." --weight=4
-
-# Backup the current version of the app
-ynh_backup_before_upgrade
-ynh_clean_setup () {
-	# restore it if the upgrade fails
-	ynh_restore_upgradebackup
-}
-# Exit if an error occurs during the execution of the script
-ynh_abort_if_errors
+if ! ynh_permission_exists --permission="admin"; then
+	# Create the required permissions
+	ynh_permission_create --permission="admin" --url="/admin" --allowed=$admin
+fi
 
 #=================================================
-# CHECK THE PATH
+# CREATE DEDICATED USER
 #=================================================
+ynh_script_progression --message="Making sure dedicated system user exists..."
 
-# Normalize the URL path syntax
-path_url=$(ynh_normalize_url_path --path_url=$path_url)
+# Create a dedicated user (if not existing)
+ynh_system_user_create --username=$app --home_dir=$final_path
 
-#=================================================
-# STANDARD UPGRADE STEPS
 #=================================================
 # DOWNLOAD, CHECK AND UNPACK SOURCE
 #=================================================
-ynh_script_progression --message="Upgrading source files..." --weight=4
 
 if [ "$upgrade_type" == "UPGRADE_APP" ]
 then
-	# Download, check integrity, uncompress and patch the source from app.src
-	# In case of a new version, the url change from http://download.dotclear.org/latest/dotclear-X.X.X.tar.gz to http://download.dotclear.org/attic/dotclear-X.X.X.tar.gz
-	src_url=$(grep 'SOURCE_URL=' "../conf/app.src" | cut -d= -f2-)
-	if ! curl --output /dev/null --silent --head --fail "$src_url"; then
-		ynh_replace_string "latest" "attic" ../conf/app.src
-	fi
+	ynh_script_progression --message="Upgrading source files..."
```

Comment:
```suggestion
	# Download, check integrity, uncompress and patch the source from app.src
	# In case of a new version, the url change from http://download.dotclear.org/latest/dotclear-X.X.X.tar.gz to http://download.dotclear.org/attic/dotclear-X.X.X.tar.gz
	src_url=$(grep 'SOURCE_URL=' "../conf/app.src" | cut -d= -f2-)
	if ! curl --output /dev/null --silent --head --fail "$src_url"; then
		ynh_replace_string "latest" "attic" ../conf/app.src
	fi
	ynh_script_progression --message="Upgrading source files..."
```